### ISSUE TITLE
EnergyDirection needs PhET-iO serialization like an Enumeration, not a Vector2

### ISSUE THREAD

**Issue by zepumph** [phetsims/greenhouse-effect#86]
I discovered a bug while debugging the issues around https://github.com/phetsims/greenhouse-effect/issues/80. There are a couple of spots where we are using object equality to check again Vector2 positioning. In PhET-iO though, these references are being tossed out and recreated as new Vector2 instances that have different values. To solve this, there are a couple of solutions.

1. (the easiest), alter the fromStateObject or applyState usages to swap out the deserialized vector2 for the right instance of the immutable reference.
1. Everywhere in the code, just use `Vector2.equals` instead of `===`.
2. Create an Enumeration class that maps to the Vector2 instances you would like.
3. (hardest) we could work on instrumenting `ImmutableVector2` for this purpose, to give us a ground truth instance throughout our work.

I'll start with (1) because it will be quite simple to do, and we can revamp the strategy as we go.

**Comment by zepumph** [comment]
Ok. I went with the easy solution, and it worked really well. It fixed #80 because the reference-equality in the conditional in `SpaceEnergySink.interactWithEnergy` was being respected.

@jbphet, I don't think investigating any other strategy for serializing this is worth doing. But let me know if you don't like the above commit. Please review.

Please note that EnergyDirection still feels like a ticking time bomb to me, it would be easy to use it somewhere else and serialize it as a Vector2 again. For example, this could be problematic if you ever tried to serialize `Substance` (which includes EnergyDirection as part of it). Up to you if we should make it more strongly typed (instead of Vector2).

**Comment by jessegreenberg** [comment]
This change was cherry-picked into the 1.0 branch as part of https://github.com/phetsims/greenhouse-effect/issues/80

**Comment by jbphet** [comment]
@zepumph said:

> "Please note that EnergyDirection still feels like a ticking time bomb to me[.]

I'm not too keen on time bombs in my code, so while it seems fine to leave it as is in the 1.0 release, let's see if we can improve things on master.

@zepumph - Would things work significantly better if `EnergyDirection` were a true enumeration?  I tried to use `RichEnumeration` for it, but it asserted saying, "rich enumeration values cannot provide their own toString".  Perhaps we could remove this restriction or make `Vector2` a supported exception to that rule.  If not that, I'd be okay to make `EnergyDirection` a simple enumeration and have a constant of type `Map` for mapping the enum to a vector when needed.  I think either of these would solve the references-become-their-own-instances-on-deserialization problem.  How do these ideas sound, and does either seem better than the other to you?

**Comment by zepumph** [comment]
I would go with the simple enum, and see how far that gets us. If we need to rope in JO or SR to accomplish this I think it will be more hard than it's worth.

How about something like this:

```diff
Index: js/common/model/EnergyDirection.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/common/model/EnergyDirection.js b/js/common/model/EnergyDirection.js
--- a/js/common/model/EnergyDirection.js	(revision d0815a32345ee7e04a46096abe3749810ec60f82)
+++ b/js/common/model/EnergyDirection.js	(date 1633647724944)
@@ -11,12 +11,26 @@
  */
 
 import Vector2 from '../../../../dot/js/Vector2.js';
+import Enumeration from '../../../../phet-core/js/Enumeration.js';
 import greenhouseEffect from '../../greenhouseEffect.js';
 
-const EnergyDirection = {
-  UP: new Vector2( 0, 1 ),
-  DOWN: new Vector2( 0, -1 )
-};
+
+const UP_VECTOR = new Vector2( 0, 1 );
+const DOWN_VECTOR = new Vector2( 0, -1 );
+
+const EnergyDirection = Enumeration.byKeys( [ 'UP', 'DOWN' ], {
+  beforeFreeze: EnergyDirection => {
+    EnergyDirection.toVector = enumValue => {
+      if ( enumValue === EnergyDirection.UP ) {
+        return UP_VECTOR;
+      }
+      else if ( enumValue === EnergyDirection.DOWN ) {
+        return DOWN_VECTOR;
+      }
+      return assert && assert( false, 'unsupported enumValue' );
+    };
+  }
+} );
 
 greenhouseEffect.register( 'EnergyDirection', EnergyDirection );
 export default EnergyDirection;

```

**Comment by jbphet** [comment]
Yes, that's exactly the sort of thing I was thinking of too.  Thanks, I'll go with something like this.
